### PR TITLE
improve handling of seeding time in `simulate_infections()`

### LINKED ISSUE #626: results of `simulate_infections` starts with unexpected number of infections
Summary:
When running `simulate_infections` starting with e.g. 100 infections, the results start with quite a different number (214 in the example below). This is not documented and probably not what a user expects.

``` r
library("EpiNow2")
#> 
#> Attaching package: 'EpiNow2'
#> The following object is masked from 'package:stats':
#> 
#>     Gamma

R <- data.frame(
  date = seq.Date(as.Date("2023-01-01"), length.out = 5, by = "day"),
  R = c(rep(1.2, 5))
)

sim <- simulate_infections(
  R = R,
  initial_infections = 100,
  generation_time = generation_time_opts(
    fix_dist(example_generation_time)
  ),
  delays = delay_opts(fix_dist(example_reporting_delay)),
  obs = obs_opts(family = "poisson")
)

sim[variable == "infections"]
#>      variable       date    value
#>        <char>     <Date>    <num>
#> 1: infections 2023-01-01 214.1121
#> 2: infections 2023-01-02 225.6793
#> 3: infections 2023-01-03 237.8341
#> 4: infections 2023-01-04 250.6285
#> 5: infections 2023-01-05 264.1041
```

<sup>Created on 2024-03-25 with [reprex v2.1.0](https://reprex.tidyverse.org)</sup>

### PR DESCRIPTION
**Pull request by sbfnk**
## Description

This PR closes #626.

It adds a `seeding_time` argument, adjusts the initial number of infections, and improves the documentation.

Using the example from the linked issue now yields:
``` r
library("EpiNow2")
#> 
#> Attaching package: 'EpiNow2'
#> The following object is masked from 'package:stats':
#> 
#>     Gamma

R <- data.frame(
  date = seq.Date(as.Date("2023-01-01"), length.out = 5, by = "day"),
  R = c(rep(1.2, 5))
)

sim <- simulate_infections(
  R = R,
  initial_infections = 100,
  generation_time = generation_time_opts(
    fix_dist(example_generation_time)
  ),
  delays = delay_opts(fix_dist(example_reporting_delay)),
  obs = obs_opts(family = "poisson")
)

sim[variable == "infections"]
#>      variable       date    value
#>        <char>     <Date>    <num>
#> 1: infections 2023-01-01 103.7012
#> 2: infections 2023-01-02 109.0943
#> 3: infections 2023-01-03 114.6880
#> 4: infections 2023-01-04 120.5479
#> 5: infections 2023-01-05 126.6988
```

<sup>Created on 2024-03-26 with [reprex v2.1.0](https://reprex.tidyverse.org)</sup>

It starts with 100 infections as expected and the first data point shown is the number of infections occurring on the following day.

## Initial submission checklist

- [X] My PR is based on a package issue and I have explicitly linked it.
- [X] I have tested my changes locally (using `devtools::test()` and `devtools::check()`).
- [X] I have added or updated unit tests where necessary.
- [X] I have updated the documentation if required and rebuilt docs if yes (using `devtools::document()`).
- [X] I have followed the established coding standards (and checked using `lintr::lint_package()`).
- [X] I have added a news item linked to this PR.

## After the initial Pull Request 

- [x] I have reviewed Checks for this PR and addressed any issues as far as I am able.

### CODE REVIEW

File: NEWS.md [hunk 1]
```diff
@@ -40,6 +40,7 @@
 * Fixed broken links in the README. By @jamesmbaazam in #617 and reviewed by @sbfnk.
 * Replaced descriptions and plot labels to be more general and clearer. By @sbfnk in #621 and reviewed by @jamesmbaazam.
 * Argument choices have been moved into default arguments. By @sbfnk in #622 and reviewed by @seabbs.
+* An argument to change the seeding time has been added to `simulate_infections`, and the documentation improved. By @sbfnk in .
```
Comment (jamesmbaazam) [inline NEWS.md:43]:
```suggestion
* `simulate_infections()` gained the argument `seeding_time` to change the seeding time. Additionally, the documentation was improved. By @sbfnk in #627 and reviewed by @jamesmbaazam.
```

File: R/simulate_infections.R [hunk 2]
```diff
@@ -16,20 +16,34 @@
...
 #'   numerical vector of length specified in [obs_opts()] as `week_length`
 #'   (default: 7) if `week_effect` is set to TRUE. Each element of the vector
 #'   gives the weight given to reporting on this day (normalised to 1).
 #'   The default is `NULL`.
 #' @param estimates deprecated; use [forecast_infections()] instead
+#' @param seeding_time Integer; the number of days before the first time point
+#'   of `R`; default is `NULL`, in which case it is set to the maximum of the
+#'   generation time. The minimum is 1 , i.e. the first reproduction number
+#'   given applies on the day after the index cases given by
+#'   `initial_infections`. If the generation time of longer than 1 day on
```
Comment (jamesmbaazam) [inline R/simulate_infections.R:33]:
```suggestion
#'   `initial_infections`. If the generation time is longer than 1 day on
```

**Comment by jamesmbaazam** [comment]
Sorry, I'm yet to review the part when `seeding_time` is applied. I'm in a workshop today so have limited capacity.

**Review by jamesmbaazam — APPROVED** [review]
LGTM.

**Comment by jamesmbaazam** [comment]
Oopsie! I was reviewing it at the same time as you were merging.

**Comment by sbfnk** [comment]
> Oopsie! I was reviewing it at the same time as you were merging.

That was the merge queue automatically merging as soon as you approved.

**Comment by jamesmbaazam** [comment]
Oh, cool! It happened so fast, I thought there was a clash.

**Pull request merged**